lib/puppet/util/windows/taskscheduler2.rb

```diff
@@ -0,0 +1,707 @@
+#require 'puppet/util/windows/error'
+#require 'win32ole' # TODO should guard this on Windows only
+
+# The TaskScheduler2 class encapsulates taskscheduler settings and behavior using the v2 API
+# https://msdn.microsoft.com/en-us/library/windows/desktop/aa383600(v=vs.85).aspx
+
+# @api private
+class Puppet::Util::Windows::TaskScheduler2
+
+  # include Puppet::Util::Windows::String
+
+  # TODO May not need this if no COM
+  # require 'ffi'
+  # extend FFI::Library
+
+  # The error class raised if any task scheduler specific calls fail.
+  class Error < Puppet::Util::Windows::Error; end
+
+  # private
+
+  # # TODO needed?
+  # class << self
+  #   attr_accessor :com_initialized
+  # end
+
+  public
+
+  # The name of the root folder for tasks
+  ROOT_FOLDER = '\\'
+  
+  # https://msdn.microsoft.com/en-us/library/windows/desktop/aa378137(v=vs.85).aspx
+  S_OK = 0
```

Comment:
This is already defined elsewhere FWIW - https://github.com/puppetlabs/puppet/blob/3a5dfed0fef8fc5cb1b6ec168766115de3e8f23b/lib/puppet/util/windows/com.rb#L8

Reply:
Ok I can use that instead